### ISSUE TITLE
Document how to execute async functions in Node 8+

### ISSUE THREAD

**Issue by jeffhollan** [Azure/azure-functions-nodejs-worker#94]
You can execute async functions in Node 8 today, but means you need to:

1. Add the `async ()` keyword before the method parameters
1. Don't use the `context.done()` method (I assume) as it throws an error and appears to work fine without it.

@asavaritayal to help assign this.  I'm happy to help or own this but wanted to make sure we were tracking somewhere.  Feel free to move to right spot

**Comment by jeffhollan** [comment]
@ColbyTresness as well - not sure who is managing the Node tasks 😃

**Comment by mhoeger** [comment]
@jeffhollan - thanks!! a small note, i think it would be better to leave out mention of node version in the verbage. we added node 10 support (not yet released though) and we're moving towards deprecating node 8 in favor of 10.

**Comment by mhoeger** [comment]
Related to https://github.com/Azure/Azure-Functions/issues/431

**Comment by paulbatum** [comment]
@jeffhollan @ColbyTresness should we turn this into a task to update the v2 templates to use async? If so we could move this issue to the templates repo. Would be nice as it would naturally demonstrate to users that they can use async and help cases where users forget to call context.done().

**Comment by jeffhollan** [comment]
Yes I have started using by default and haven't seen any bad side-effects. I think it makes sense to move to the templates repo as a v2 must have

**Comment by jeffhollan** [comment]
[Opened a PR](https://github.com/Azure/azure-functions-templates/pull/718)

**Comment by paulbatum** [comment]
@jeffhollan @ColbyTresness @mhoeger can we close this?

**Comment by ColbyTresness** [comment]
Lets leave it for a bit longer (1 week max) since this is something we've touched but haven't done a final pass through yet. I'll close this when that happens.

**Comment by mhoeger** [comment]
@ColbyTresness - how is [this](https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-node#contextdone-method)? or does async deserve its own bucket?

**Comment by ColbyTresness** [comment]
@mhoeger Sorry about the delay. I think what we have is a good start, but it doesn't seem sufficient: the only place we mention async in our JavaScript developer guide is in the context.done() section, which doesn't 100% make sense since using async means they don't need context.done(), and hence wouldn't look there. Upon further examination, I think this is important enough information to add as a paragraph in the intro section of the JS Dev guide. Thoughts?

**Comment by egiray** [comment]
Hi, so where is the document? Still I can't use async function and it doesn't behave normal.

**Comment by paulbatum** [comment]
Just for anyone that finds this issue and is looking for the updated docs, they are here:
https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-node#exporting-an-async-function
